freqtrade/exchange/exchange.py

```diff
@@ -1486,12 +1510,25 @@ def get_historic_trades(self, pair: str,
         :returns List of trade data
         """
         if not self.exchange_has("fetchTrades"):
-            raise OperationalException("This exchange does not suport downloading Trades.")
+            raise OperationalException("This exchange does not support downloading Trades.")
 
         return asyncio.get_event_loop().run_until_complete(
             self._async_get_trade_history(pair=pair, since=since,
                                           until=until, from_id=from_id))
 
+    def get_interest_rate(self, pair: str, open_rate: float, is_short: bool) -> float:
+        # TODO-lev: implement
+        return 0.0005
+
+    def set_leverage(self, pair, leverage):
```

Comment:
I think this whole method will need further investigation.
In my understanding, it's a "binance only" feat - which means we will not implement it in the main exchange class - but call it as part of the "create_lev_order" stuff in the binance subclass.

having it in the parent (this) class will only make sense if it's something that's generalized - which i doubt.

Reply:
If it's binance only, wouldn't it be an sapi method?

Reply:
you can't take this assumption from the ccxt method name.

even if it's not binance only, but impacts binance and some random exchange we never heared of (none of the ones we support) - it would be wrong to have it as part of the global interface (and have freqtradebot.py call it).

To me, this is a implementation detail per exchange - which means - we call `create_order(...,leverage=3)` - and the respective exchange-class then makes sure this is followed through.
This CAN mean calling `.set_leverage(3)` - but it can also mean `self._api.create_order(..., {params={'leverage': 3})`.
As this is not uniform - i think it's better to leave this decision to the exchange - as it'll be simpler to adjust should some other exchange require a 3rd call for whatever reason.


---
either way - _implementing_  it here is wrong 100%.

Assuming it's used by binance and FTX (will need to be analyzed) - you could still not call the ccxt method  in `exchange.py` - as that would mean ALL exchanges MUST support this call in ccxt - which i'm sure they do not (kraken does not use this - so this is a bug for kraken that will cause a crash).